### PR TITLE
e2e: free resources on successful runs

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 

because we share a machine to run e2e tests, if we don't free resources, a lot of processes will run in the machine even if no e2e tests are running.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change
 - Has Go code change
 - Has CI related scripts change
 - Has documents change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note

 ```

### CODE REVIEW

File: tests/cmd/e2e/main.go [hunk 1]
```diff
@@ -188,22 +188,27 @@ func main() {
 		testBasic(&wg, cluster1)
 		testHostNetwork(&wg, cluster1)
 		oa.CheckDataRegionDisasterToleranceOrDie(cluster1)
+		oa.CleanTidbClusterOrDie(cluster1)
```
Comment (weekface) [inline tests/cmd/e2e/main.go]:
```suggestion
		go oa.CleanTidbClusterOrDie(cluster1)
```

clean a cluster may need a few minutes. Can do it in a goroutine.

Comment (cofyc) [reply tests/cmd/e2e/main.go]: it's already in the goroutine of this task (see previous code hidden by GitHub)

Comment (cofyc) [reply tests/cmd/e2e/main.go]: what about deleting the `e2e` namespace at the end of main function and wait all resources are freed?

Comment (weekface) [reply tests/cmd/e2e/main.go]: Yes, but it increases the time of that goroutine. Failed to clean is not a failure of the whole e2e test.

Comment (weekface) [reply tests/cmd/e2e/main.go]: Just a suggestion, not a necessity.

**Comment by cofyc** [comment]
/run-e2e-in-kind
rerun after rebase

**Pull request merged**